docs/theory/diffusion/diffusion.rst

```diff
@@ -0,0 +1,715 @@
+This way, longer tracks contribute more to the estimate than short tracks which is beneficial for both accuracy and precision.
+The weighted average is computed as :cite:`vestergaard2014optimal`:
+
+.. math::
+
+    \overline{\hat{D}} = \frac{\sum_{m=1}^M N_m \hat{D}_m}{\sum_{m=1}^M N_m}
+
+Here :math:`M` is the number of tracks, :math:`D_m` corresponds to the diffusion constant of track :math:`m` while :math:`N_m` corresponds to the number of data points contributing to its estimate.
+The associated variance of the weighted mean is given by:
+
+.. math::
+
+    \overline{\hat{D}} = \frac{\sum_{m=1}^M N_m \left(\hat{D}_m - \overline{\hat{D}}\right)^2}{\left(M - 1\right) \sum_{m=1}^M N_m}
+
+To check how the performance of the ensemble estimators compares to individual track estimation, the estimation procedure becomes a bit more complex.
+Instead of simulating a single track many times, we simulate a collection of tracks many times and perform estimation on this.
+Unfortunately, this means that these notebook cells also take more time to evaluate::
+
+    def simulate_tracks(snr):
+        """Function used to simulate tracks"""
+        params = snr_to_diffusion_parameters(snr)
+        tracks = [
+            lk.simulation.simulate_diffusive_tracks(**params, steps=8, num_tracks=10)
+            for _ in range(100)
+        ]
+        return tracks, params["diffusion_constant"]
+
+    methods = {
+        "cve single": lambda list_of_tracks: [t.estimate_diffusion("cve") for tracks in list_of_tracks for t in tracks],
+        "cve ensemble": lambda list_of_tracks: [t.ensemble_diffusion("cve") for t in list_of_tracks],
+    }
+
+    snrs = 10**np.arange(-1, 1.1, 0.125)
+    results, variances = test_estimation_methods(simulate_tracks, snrs, methods=methods)
+    for method, estimates in results.items():
+        plot_accuracy(snrs, estimates, variance=variances[method], label=method);
+
+    plt.ylim([-5, 6])
+
+.. image:: figures/cve_ensemble.png
+
+Here we simulated `10` tracks with `8` steps per track.
+As expected, the ensemble CVE results in a far lower variances (each estimate uses more information).
+The uncertainty estimate Pylake returns on the individual CVEs is a little conservative this time (vertical lines), but the ensemble uncertainty estimate is pretty accurate.
+
+If we can safely assume that the localization uncertainty is constant then it is possible to improve the diffusion estimates by computing an ensemble averaged localization error first and then using that estimate when determining the per-track CVE.
+Let's define a function that does just this and compare it to the single step CVE::
+
+    def cve_single_using_ensemble_loc_uncertainty(tracks):
+        ensemble_estimate = tracks.ensemble_diffusion("cve")
+        return tracks.estimate_diffusion(
+            method="cve",
+            # Plug in the ensemble localization uncertainty estimates
```

Comment:
comment inside the function call is a bit weird, I'd move it above

Reply:
Moved.